doc/guides/random_numbers.rst

```diff
@@ -1,523 +1,483 @@
-N}. Random *deviate* generators, on the other hand, provide random
-numbers drawn from a range of distributions, such as the normal or
-binomial distributions. In most cases, you will be using random deviate
-generators. They are in particular used to initialize properties during
-network construction, as described in the sections :ref:`changes-nest>`
-and :ref:`examples-rng` below.
+   Due to the significant changes in random number architecture between NEST 2
+   and NEST 3, it is not possible to re-create NEST 2 simulations with identical
+   random number sequences in NEST 3. Thus, spike patterns will differ in detail
+   between simulations of the same model with NEST 2 and NEST 3, but shall agree
+   as long as suitable quantities, such as firing rates, correlations, or spectra
+   are compared across multiple simulations.
 
-.. _changes-nest:
 
-Changes in random number generation in NEST 2.4
------------------------------------------------
+.. admonition:: Compiler dependency
 
-Random deviate generation has become significantly more powerful in NEST
-2.4, to fully support randomization of connections parameters offered by
-the revised ``Connect`` function, as described in :doc:`Connection
-Management <connection_management>` and illustrated by the
-:ref:`examples-rng` below. We have also made minor
-changes to make to achieve greater similarity between NEST, PyNN, and
-NumPy. For most users, these changes only add new features. Only
-existing scripts using
+   NEST 3 uses random generators provided by the C++ Standard Library to
+   obtain random deviates, i.e., random numbers following different distributions
+   such as normal, binomial, or Poissonian. Different versions of the C++
+   Standard Library use different algorithms to generate these deviates.
+   Therefore, simulation results can differ in detail depending on the compiler
+   and the C++ Standard Library implementation used. Note in particular that
+   GCC and Clang come with different library implementations.
 
--  ``uniformint``
--  ``normal_clipped``, ``normal_clipped_left``, ``normal_clipped_right``
+   As for the difference between NEST 2 and NEST 3, simulations based on
+   different compilers/standard libraries shall agree as long as suitable
+   quantities are compared using suitable statistical tests.
 
-generators from NEST 2.2 need to be adapted as detailed below.
 
-The changes are as follows:
+.. _working_with_rngs:
 
--  Uniform integer generator
+Working with randomness in NEST
+-------------------------------
 
-   -  renamed from ``uniformint`` to ``uniform_int``
-   -  parameters renamed to ``low`` and ``high``
-   -  returns uniformly distributed integers from
-      ``{low, low+1, …, high}``
+NEST 3 makes randomness easy: Setting a single seed gives you full
+control over randomness throughout a simulation, and you can choose seed values
+at liberty. NEST's powerful mechanism to :ref:`parameterize nodes and synapses <REFERENCE NEEDED>`
```

Comment:
Remember this reference!

Reply:
@heplesser Should it be `param_ex` for the section in the [NEST 2 to 3 guide](https://nest-simulator.readthedocs.io/en/latest/guides/nest2_to_nest3/nest2_to_nest3_overview.html#parametrization), or `random_examples` for [the examples below in this guide](https://github.com/nest/nest-simulator/blob/5af666418730e3a0a7f7a1c0f717d19dd23a6434/doc/guides/random_numbers.rst#examples-of-using-randomness)?